app/celery/process_delivery_status_result_tasks.py

```diff
@@ -265,3 +281,221 @@ def sms_status_update(
+        event_timestamp (str | None, optional): Timestamp the Pinpoint event came in. Defaults to None.
+        event_in_seconds (int, optional): How long since initial event. Defaults to 300
+
+    Raises:
+        NonRetryableException: Unable to update SMS retry count or update the notification
+    """
+
+    # avoid circular import
+    from app.notifications.process_notifications import send_notification_to_queue
+
+    notification = _get_notification(sms_status.reference, sms_status.provider, event_timestamp, event_in_seconds)
+
+    current_app.logger.info(
+        'Entering %s retryable failure logic to process event | reference: %s | notification_id: %s | current_status: %s | current_status_reason: %s | event_sms_status: %s | event_sms_status_reason: %s',
+        sms_status.provider,
+        sms_status.reference,
+        notification.id,
+        notification.status,
+        notification.status_reason,
+        sms_status.status,
+        sms_status.status_reason,
+    )
+
+    notification_retry_id = f'notification-carrier-sms-retry-count-{notification.id}'
+
+    try:
+        retry_count_redis_ttl = int(CARRIER_SMS_MAX_RETRY_WINDOW.total_seconds())
+        retry_count = update_sms_retry_count(notification_retry_id, ttl=retry_count_redis_ttl)
+    except Exception:
+        raise NonRetryableException('Unable to update SMS retry count')
+
+    if can_retry_sms_request(
+        notification.status, retry_count, CARRIER_SMS_MAX_RETRIES, notification.sent_at, CARRIER_SMS_MAX_RETRY_WINDOW
+    ):
+        retry_delay = get_sms_retry_delay(retry_count)
+
+        # need to roll notification status back to 'created' for requeue to work
+        # TODO: re-add exception test
+        try:
+            notification: Notification = dao_update_sms_notification_status_to_created_for_retry(
+                notification_id=notification.id,
+                notification_type=notification.notification_type,
+                cost_in_millicents=notification.cost_in_millicents + sms_status.price_millicents,
+            )
+
+            current_app.logger.info(
+                'Notification updated prior to requeue attempt | notification_id: %s | notification_status: %s | cost_in_milicents %s',
+                notification.id,
+                notification.status,
+                notification.cost_in_millicents,
+            )
+            statsd_client.incr(f'clients.sms.{sms_status.provider}.delivery.status.{sms_status.status}')
+        except Exception:
+            statsd_client.incr(f'clients.sms.{sms_status.provider}.status_update.error')
+            raise NonRetryableException('Unable to update notification')
+
+        current_app.logger.info(
+            'Attempting %s requeue | notification_id: %s | retry_delay: %s seconds | retry_count: %s',
+            sms_status.provider,
+            notification.id,
+            str(retry_delay),
+            str(retry_count),
```

Comment:
fyi - the %s will convert a non-string into a string, so converting to a string is not necessary.
https://docs.python.org/3.10/library/stdtypes.html#old-string-formatting